src/_P064_APDS9960.ino

```diff
@@ -182,25 +286,26 @@ boolean Plugin_064(byte function, struct EventStruct *event, String& string)
 
         if (1)
         {
-          uint8_t proximity_data = 0;
-          PLUGIN_064_pds->readProximity(proximity_data);
-          UserVar[event->BaseVarIndex + 1] = (float)proximity_data;
-
-          uint16_t ambient_light = 0;
-          PLUGIN_064_pds->readAmbientLight(ambient_light);
-          UserVar[event->BaseVarIndex + 2] = (float)ambient_light;
-
-          /*
-          uint16_t red_light = 0;
-          uint16_t green_light = 0;
-          uint16_t blue_light = 0;
-          PLUGIN_064_pds->readRedLight(red_light);
-          PLUGIN_064_pds->readGreenLight(green_light);
-          PLUGIN_064_pds->readBlueLight(blue_light);
-          UserVar[event->BaseVarIndex + 3] = (float)red_light;
-          UserVar[event->BaseVarIndex + 4] = (float)green_light;
-          UserVar[event->BaseVarIndex + 5] = (float)blue_light;
-          */
+          if (PCONFIG(1) == PLUGIN_MODE_GPL_064) { // Gesture/Proximity/ALS mode
+            uint8_t proximity_data = 0;
+            PLUGIN_064_pds->readProximity(proximity_data);
+            UserVar[event->BaseVarIndex + 1] = (float)proximity_data;
+
+            uint16_t ambient_light = 0;
+            PLUGIN_064_pds->readAmbientLight(ambient_light);
+            UserVar[event->BaseVarIndex + 2] = (float)ambient_light;
+
+          } else {
+            uint16_t red_light = 0;
+            uint16_t green_light = 0;
+            uint16_t blue_light = 0;
+            PLUGIN_064_pds->readRedLight(red_light);
+            PLUGIN_064_pds->readGreenLight(green_light);
+            PLUGIN_064_pds->readBlueLight(blue_light);
+            UserVar[event->BaseVarIndex + 0] = (float)red_light;
+            UserVar[event->BaseVarIndex + 1] = (float)green_light;
+            UserVar[event->BaseVarIndex + 2] = (float)blue_light;

```

Comment:
Please use `static_cast<float>(...)` instead of C-style cast.
But I don't think it is needed anyway, as the type of `UserVar` is float.

Reply:
My C++ knowledge is lacking in this area, I'm learning on the job :wink:

Reply:
Well the compiler will cast them if it can cast them.
An int can be casted to a float, but it is also not bad to cast them explicitly.
However, an old fashioned C-style cast is not the best way to do. Better use `static_cast<float>(red_light)` instead of `(float)red_light`

Reply:
I added those static_casts in my 'improvements' commit